File: DrivingSchoolBackend/DrivingSchoolApi/Models/Question.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace DrivingSchoolApi.Models
{
    public class Question: BaseEntity
	{
		public required string Title {get; set;}
        public required string Text {get; set;}
        // Todo add theme object?
        public required string[] Answers {get; set;}
        public required int AnswerIndex {get; set;}
        public string[]? Themes {get; set;}
        public required string ImageUri {get; set;}
        public string? Explanation {get; set;}
        [JsonIgnore]
        public List<User>? Users { get; set; } 
    }
}

File: DrivingSchoolFrontEnd/src/models/Question.ts
import { QuestionDTO } from "../interfaces/interfaces";
import { QuizzSummaryElement } from "./QuizzSummaryElement";

export class Question {
    id?: number;
    title: string;
    text: string;
    answers: string[];
    answerIndex: number;
    themes: string[];
    imageUri: string;

    constructor(title: string, text: string, answers: string[], answerIndex: number, themes: string[], imageUri: string) {
        this.title = title;
        this.text = text;
        this.answers = answers;
        this.answerIndex = answerIndex;
        this.themes = themes;
        this.imageUri = imageUri;
      }
    
      static fromDTO(dto: QuestionDTO): Question {
        return new Question(dto.title, dto.text, dto.answers, dto.answerIndex, dto.themes, dto.imageUri);
      }

}
